include/boost/geometry/algorithms/line_interpolate_point.hpp

```diff
@@ -0,0 +1,402 @@
+#include <boost/geometry/core/tags.hpp>
+
+#include <boost/geometry/geometries/concepts/check.hpp>
+
+#include <boost/geometry/algorithms/assign.hpp>
+#include <boost/geometry/algorithms/length.hpp>
+#include <boost/geometry/strategies/default_strategy.hpp>
+#include <boost/geometry/strategies/line_interpolate_point.hpp>
+
+namespace boost { namespace geometry
+{
+
+
+#ifndef DOXYGEN_NO_DETAIL
+namespace detail { namespace interpolate_point
+{
+
+struct convert_and_push_back
+{
+    template <typename Range, typename Point>
+    inline void apply(Point const& p, Range& range)
+    {
+        typename boost::range_value<Range>::type p2;
+        geometry::detail::conversion::convert_point_to_point(p, p2);
+        range::push_back(range, p2);
+    }
+};
+
+struct convert_and_assign
+{
+    template <typename Point1, typename Point2>
+    inline void apply(Point1& p1, Point2& p2)
+    {
+        geometry::detail::conversion::convert_point_to_point(p1, p2);
+    }
+
+};
+
+
+/*!
+\brief Internal, calculates interpolation point of a linestring using iterator pairs and
+    specified strategy
+*/
+template <typename Policy>
+struct range
+{
+    template <typename Range, typename PointType, typename Strategy>
+    static inline void apply(Range const& range,
+                             double const& fraction,
+                             PointType & p_or_mp,
+                             Strategy const& strategy)
+    {
+        Policy policy;
+
+        typedef typename boost::range_iterator<Range const>::type iterator_t;
+        typedef typename boost::range_value<Range const>::type point_t;
+
+        typedef typename select_most_precise
+            <
+                typename default_length_result<Range>::type,
+                double
+            >::type calc_t;
+
+        typedef typename Strategy::template result_type<point_t> result_type;
+        typedef boost::container::vector<result_type> vector_t;
+        typedef typename boost::range_iterator<vector_t const>::type viterator_t;
+
+        iterator_t it = boost::begin(range);
+        iterator_t end = boost::end(range);
+
+        if (it == end) // empty(range)
+        {
+            return;
+        }
+        if (fraction == 0)
+        {
+            policy.apply(*it, p_or_mp);
+            return;
+        }
+        if (fraction == 1)
+        {
+            policy.apply(*(end-1), p_or_mp);
+            return;
+        }
+            
+        // compute lengths of segments and their sum
+        calc_t tot_len = 0;
+        vector_t lengths;
+        iterator_t prev = it++;
+        do {
+            result_type len = strategy.compute(*prev, *it);
+            lengths.push_back(len);
+            tot_len += len.distance;
+            prev = it++;
+        } while (it != end);
+
+        //compute interpolation point(s)
+        calc_t repeated_fraction = fraction;
+        calc_t prev_fraction = 0;
+        calc_t cur_fraction = 0;
+        unsigned int i = 1;
```

Comment:
I'm not sure if `unsigned int` is a proper type here since on x64 it will still be 32b type. What do you think about changing it to `std::size_t`? It's for counting fractions, not elements of range but since fraction is typically in range [0, 1] and theoretically there may be more points in the output than it is in the input then using at least as big type to count fractions as we use to count elements has sense.